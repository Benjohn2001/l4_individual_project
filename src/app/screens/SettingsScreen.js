import React from "react";
import { View, Text, SafeAreaView } from "react-native";
import { getAuth } from "firebase/auth";
import RowItem from "../components/RowItem";

function SettingsScreen({ navigation }) {
  return (
    <SafeAreaView className="flex-1 bg-primaryPurple">
      <Text className="font-bold text-3xl justify-start pt-14 mx-10">
        Settings
      </Text>
      <Text className="font-bold text-2xl justify-start pt-10 mx-5 ">
        General
      </Text>
      <View className="items-center">
        <RowItem
          title="Account"
          icon="user"
          onPress={() => {
            alert("Account");
          }}
          color="black"
        />
        <RowItem
          title="Notifications"
          icon="bell"
          onPress={() => {
            alert("Notifications");
          }}
          color="black"
        />
        <RowItem
          title="Log Out"
          icon="log-out"
          onPress={() => {
            getAuth()
              .signOut()
              .then(() => {
                navigation.push("SignInScreen");
              })
              .catch((error) => {
                console.log(error);
              });
          }}
          color="red"
        />
      </View>

      <Text className="font-bold text-2xl justify-start pt-10 mx-5 ">
        Feedback
      </Text>
      <View className="items-center">
        <RowItem
          title="Send Feedback"
          icon="send"
          onPress={() => {
            alert("Send Feedback");
          }}
          color="black"
        />
        <RowItem
          title="Report a bug"
          icon="alert-triangle"
          onPress={() => {
            alert("Report a bug");
          }}
          color="black"
        />
      </View>
    </SafeAreaView>
  );
}

export default SettingsScreen;
