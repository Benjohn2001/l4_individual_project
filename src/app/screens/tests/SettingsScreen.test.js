import renderer from "react-test-renderer";
import { render, screen, fireEvent } from '@testing-library/react-native';
import SettingsScreen from '../SettingsScreen';

jest.mock('firebase/database', () => ({
    getDatabase: jest.fn().mockReturnThis(),
  }));

jest.mock('firebase/storage', () => ({
      getStorage: jest.fn().mockReturnThis(),
    }));
  
jest.mock('firebase/app', () => ({
      auth:  jest.fn().mockReturnThis(),
      initializeApp: jest.fn().mockReturnThis(),
    }));

jest.mock('firebase/auth', () => ({
    auth:  jest.fn().mockReturnThis(),
      getAuth:  () => ({
          signOut:  () => ({
            then: () => ({
                catch: jest.fn()
            })
            
          }),
        }),
    }));

test('Check SettingsScreen working correctly',()=>{

    render(
        <SettingsScreen/>
    );

    global.alert=jest.fn()
    fireEvent.press(screen.getByText('Account'));
    expect(global.alert).toHaveBeenCalledTimes(1)
    fireEvent.press(screen.getByText('Notifications'));
    expect(global.alert).toHaveBeenCalledTimes(2)
    fireEvent.press(screen.getByText('Send Feedback'));
    expect(global.alert).toHaveBeenCalledTimes(3)
    fireEvent.press(screen.getByText('Report a bug'));
    expect(global.alert).toHaveBeenCalledTimes(4)
    fireEvent.press(screen.getByText('Log Out'));

    const domTree = renderer.create(
        <SettingsScreen/>
        ).toJSON();
      expect(domTree).toMatchSnapshot();
})



