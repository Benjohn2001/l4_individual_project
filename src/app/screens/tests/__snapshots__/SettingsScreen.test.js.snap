// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check SettingsScreen working correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "backgroundColor": "#C6C4FF",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "700",
        },
        Object {
          "fontSize": 30,
          "lineHeight": 36,
        },
        Object {
          "justifyContent": "flex-start",
        },
        Object {
          "paddingTop": 56,
        },
        Object {
          "marginLeft": 40,
          "marginRight": 40,
        },
      ]
    }
  >
    Settings
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "700",
        },
        Object {
          "fontSize": 24,
          "lineHeight": 32,
        },
        Object {
          "justifyContent": "flex-start",
        },
        Object {
          "paddingTop": 40,
        },
        Object {
          "marginLeft": 20,
          "marginRight": 20,
        },
      ]
    }
  >
    General
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            Account
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            Notifications
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            Log Out
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "700",
        },
        Object {
          "fontSize": 24,
          "lineHeight": 32,
        },
        Object {
          "justifyContent": "flex-start",
        },
        Object {
          "paddingTop": 40,
        },
        Object {
          "marginLeft": 20,
          "marginRight": 20,
        },
      ]
    }
  >
    Feedback
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            Send Feedback
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            Report a bug
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
